nikola/feedutil.py

```diff
@@ -0,0 +1,463 @@
+
+    def _tzdatetime(self, dt):
+        """Get a datetime for feed."""
+        if dt.tzinfo is None:
+            dt = dt.replace(tzinfo=self.site.tzinfo)
+        return dt.astimezone(dateutil.tz.tzutc())
+
+    def _enclosure(self, post, lang):
+        """Get an enclosure infomation from a post."""
+        enclosure = post.meta('enclosure', lang)
+        if enclosure:
+            length = 0
+            url = enclosure
+            mime = mimetypes.guess_type(url)[0]
+            return url, length, mime
+
+    def get_feed_content(self, data, lang, post, enclosure_details,
+                         preview_image=None, default_image=None):
+        """Create a feed content."""
+        try:
+            doc = html5lib.html5parser.parse(data, treebuilder='lxml',
+                                             namespaceHTMLElements=False)
+            doc = fragment_fromstring(lxml.html.tostring(doc), create_parent=True)
+            doc.rewrite_links(
+                lambda dst: self.site.url_replacer(post.permalink(), dst, lang,
+                                                   'absolute'))
+            try:
+                src = None
+                if preview_image:
+                    src = preview_image
+                elif (enclosure_details and
+                      enclosure_details[2].split('/')[0] == 'image'):
+                    src = enclosure_details[0]
+                else:
+                    postdoc = html5lib.html5parser.parse(post.text(lang),
+                                                         treebuilder='lxml',
+                                                         namespaceHTMLElements=False)
+                    found = postdoc.xpath('//img')
+                    for img in found:
+                        src = img.attrib.get('src')
+                        if src:
+                            break
+                    if src is None and default_image:
+                        src = default_image
+                if src is not None and src not in data:
+                    f = lxml.etree.Element('fiture')
+                    a = lxml.etree.SubElement(f, 'a')
+                    a.attrib['href'] = post.permalink(lang, absolute=True)
+                    img = lxml.etree.SubElement(a, 'img')
+                    img.attrib['src'] = src
+                    img.attrib['alt'] = 'thumbnail'
+                    doc.insert(0, f)
+
+                data = (doc.text or '') + ''.join(
+                    [lxml.html.tostring(child, encoding='unicode')
+                     for child in doc.iterchildren()])
+            # No body there, it happens sometimes
+            except IndexError:
+                data = ''
+        except lxml.etree.ParserError as e:
+            if str(e) == "Document is empty":
+                data = ""
+            else:  # let other errors raise
+                raise(e)
+        return data
+
+    def gen_uuid(self, path):
```

Comment:
No. Remove this. The UUID is not recreatable so we cannot produce the same output twice. Use the permalink URIs as IDs as we do today.